TypeCobol/Compiler/CodeElements/CodeElement.cs

```diff
@@ -157,16 +159,36 @@ public override string ToString()
 
         public override bool Equals(object obj)
         {
-            var codeElement = obj as CodeElement;
-            if (codeElement == null)
-                return false;
-
-            return this.Line == codeElement.Line &&
-                   this.Type == codeElement.Type &&
-                   this.Column == codeElement.Column &&
-                   this.StartIndex == codeElement.StartIndex &&
-                   this.StopIndex == codeElement.StopIndex &&
-                   this.Text == codeElement.Text;
+            return Equals(obj as CodeElement);
+        }
+
+        public bool Equals(CodeElement codeElement)
+        {
+            if (Object.ReferenceEquals(this, codeElement)) return true;
```

Comment:
Can you try without overriding Equals and GetHashCode?

Reply:
It doesn't work without overriding Equals and GetHashCode.